enterprise/frontend/src/embedding-sdk/cli/snippets/nextjs-app-snippets.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Custom app snippets for the Pages Router.
+ */
+function getNextJsCustomAppSnippet({
+  resolveImport,
+}: {
+  resolveImport: (pathName: string) => string;
+}) {
+  const snippet = `
+import { AnalyticsProvider } from "${resolveImport("analytics-provider")}"
+import { EmbeddingProvider } from "${resolveImport("embedding-provider")}"
+
+import "${resolveImport("analytics.css")}"
+
+export default function CustomApp({ Component, pageProps }) {
+  return (
+    <AnalyticsProvider>
+      <EmbeddingProvider>
+        <Component {...pageProps} />
+      </EmbeddingProvider>
+    </AnalyticsProvider>
+  )
+}
+`;
+
+  return snippet.trim();
+}
+
+/**
+ * Generates a default Next.js root layout.
+ * This file should be placed as `app/layout.tsx`.
+ *
+ * @see https://nextjs.org/docs/app/api-reference/file-conventions/layout#root-layouts
+ */
+function getNextJsRootLayoutSnippet({
+  resolveImport,
+}: {
+  resolveImport: (path: string) => string;
+}) {
+  const snippet = `
+import { AnalyticsProvider } from "${resolveImport("analytics-provider")}"
+import { EmbeddingProvider } from "${resolveImport("embedding-provider")}"
+
+import "${resolveImport("analytics.css")}"
+
+export default function RootLayout({children}) {
+  return (
+    <html>
+      <body>
+        <AnalyticsProvider>
+          <EmbeddingProvider>
+            {children}
+          </EmbeddingProvider>
+        </AnalyticsProvider>
+      </body>
+    </html>
+  )
+}
+  `;
+
+  return snippet.trim();
+}
+
+export function getNextJsCustomAppOrRootLayoutSnippet({
```

Comment:
nit: Personally I'd just name this PagesWrapper or AppWrapper or something. I'd bet people would know the difference between pages and app router on top of their heads better than custom app vs root layout. I don't even know which one is for pages router and which one is for app router.

Some example suggested name would be `getNextJsPagesWrapperOrAppWrapper` or something similar.

Reply:
Another point is that this function might not belong here. This file is `nextjs-app-snippets` it should only contains stuff related to _only_ app router, not something more generic.

Reply:
Good point, there were only "custom app" in the beginning of the PR so this file was for "custom app / root layout". I'll find a less confusing name.

Reply:
fixed in 8ef475db9a97db31bbfd5c33181af10a7d5ad0b5